src/protocol/prf_sharding/mod.rs

```diff
@@ -0,0 +1,704 @@
+        Replicated::new(
+            Gf2::truncate_from(input_row.breakdown_key.left()[i]),
+            Gf2::truncate_from(input_row.breakdown_key.right()[i]),
+        )
+    });
+    let tv = BitDecomposed::decompose(num_trigger_value_bits, |i| {
+        Replicated::new(
+            Gf2::truncate_from(input_row.trigger_value.left()[i]),
+            Gf2::truncate_from(input_row.trigger_value.right()[i]),
+        )
+    });
+    assert!(bd_key.len() == num_breakdown_key_bits);
+    assert!(
+        inputs_required_from_previous_row
+            .attributed_breakdown_key_bits
+            .len()
+            == num_breakdown_key_bits
+    );
+    assert!(tv.len() == num_trigger_value_bits);
+    assert!(inputs_required_from_previous_row.saturating_sum.len() == num_saturating_sum_bits);
+
+    let share_of_one = Replicated::share_known_value(&ctx, Gf2::ONE);
+
+    // TODO: compute ever_encountered_a_source_event and attributed_breakdown_key_bits in parallel
+    let ever_encountered_a_source_event = input_row
+        .is_trigger_bit
+        .multiply(
+            &inputs_required_from_previous_row.ever_encountered_a_source_event,
+            ctx.narrow(&Step::EverEncounteredSourceEvent(row_for_user)),
+            record_id,
+        )
+        .await?
+        + &share_of_one
+        - &input_row.is_trigger_bit;
+
+    let narrowed_ctx = ctx.narrow(&Step::AttributedBreakdownKey(row_for_user));
+    let attributed_breakdown_key_bits = BitDecomposed::new(
+        try_join_all(
+            bd_key
+                .iter()
+                .zip(
+                    inputs_required_from_previous_row
+                        .attributed_breakdown_key_bits
+                        .iter(),
+                )
+                .enumerate()
+                .map(|(i, (bd_key_bit, prev_row_attributed_bd_key_bit))| {
+                    let c = narrowed_ctx.narrow(&BitOpStep::from(i));
+                    async move {
+                        let maybe_diff = input_row
+                            .is_trigger_bit
+                            .multiply(
+                                &(prev_row_attributed_bd_key_bit.clone() - bd_key_bit),
+                                c,
+                                record_id,
+                            )
+                            .await?;
+                        Ok::<_, Error>(maybe_diff + bd_key_bit)
+                    }
+                }),
+        )
+        .await?,
+    );
+
+    let did_trigger_get_attributed = input_row
+        .is_trigger_bit
+        .multiply(
+            &ever_encountered_a_source_event,
+            ctx.narrow(&Step::DidTriggerGetAttributed(row_for_user)),
+            record_id,
+        )
+        .await?;
+
+    let narrowed_ctx = ctx.narrow(&Step::AttributedTriggerValue(row_for_user));
+    let attributed_trigger_value = BitDecomposed::new(
+        try_join_all(
+            tv.iter()
+                .zip(repeat(did_trigger_get_attributed.clone()))
```

Comment:
```suggestion
                .zip(repeat(&did_trigger_get_attributed))
```

and then you don't need an extra `&` on the line 441